zk_toolbox/crates/zk_inception/src/commands/consensus.rs

```diff
@@ -0,0 +1,313 @@
+use std::{borrow::Borrow, collections::HashMap, sync::Arc};
+
+/// Consensus registry contract operations.
+/// Includes code duplicated from `zksync_node_consensus::registry::abi`.
+use anyhow::Context as _;
+use common::config::global_config;
+use config::EcosystemConfig;
+use ethers::{
+    abi::Detokenize,
+    contract::{FunctionCall, Multicall},
+    middleware::{Middleware, NonceManagerMiddleware, SignerMiddleware},
+    providers::{Http, JsonRpcClient, PendingTransaction, Provider, RawCall as _},
+    signers::{LocalWallet, Signer as _},
+    types::{BlockId, H256},
+};
+use xshell::Shell;
+use zksync_consensus_crypto::ByteFmt;
+use zksync_consensus_roles::{attester, validator};
+
+use crate::{messages, utils::consensus::parse_attester_committee};
+
+#[allow(warnings)]
+mod abi {
+    include!(concat!(env!("OUT_DIR"), "/consensus_registry_abi.rs"));
+}
+
+fn decode_attester_key(k: &abi::Secp256K1PublicKey) -> anyhow::Result<attester::PublicKey> {
+    let mut x = vec![];
+    x.extend(k.tag);
+    x.extend(k.x);
+    ByteFmt::decode(&x)
+}
+
+fn decode_weighted_attester(
+    a: &abi::CommitteeAttester,
+) -> anyhow::Result<attester::WeightedAttester> {
+    Ok(attester::WeightedAttester {
+        weight: a.weight.into(),
+        key: decode_attester_key(&a.pub_key).context("key")?,
+    })
+}
+
+fn encode_attester_key(k: &attester::PublicKey) -> abi::Secp256K1PublicKey {
+    let b: [u8; 33] = ByteFmt::encode(k).try_into().unwrap();
+    abi::Secp256K1PublicKey {
+        tag: b[0..1].try_into().unwrap(),
+        x: b[1..33].try_into().unwrap(),
+    }
+}
+
+fn encode_validator_key(k: &validator::PublicKey) -> abi::Bls12381PublicKey {
+    let b: [u8; 96] = ByteFmt::encode(k).try_into().unwrap();
+    abi::Bls12381PublicKey {
+        a: b[0..32].try_into().unwrap(),
+        b: b[32..64].try_into().unwrap(),
+        c: b[64..96].try_into().unwrap(),
+    }
+}
+
+fn encode_validator_pop(pop: &validator::ProofOfPossession) -> abi::Bls12381Signature {
+    let b: [u8; 48] = ByteFmt::encode(pop).try_into().unwrap();
+    abi::Bls12381Signature {
+        a: b[0..32].try_into().unwrap(),
+        b: b[32..48].try_into().unwrap(),
+    }
+}
+
+#[derive(clap::Subcommand, Debug)]
+pub enum Command {
+    /// Sets the attester committee in the consensus registry contract to
+    /// `consensus.genesis_spec.attesters` in general.yaml.
+    SetAttesterCommittee,
+    /// Fetches the attester committee from the consensus registry contract.
+    GetAttesterCommittee,
+}
+
+/// Collection of sent transactions.
+#[derive(Default)]
+pub struct TxSet(Vec<(H256, &'static str)>);
```

Comment:
recently we introduced this functionality 
https://github.com/matter-labs/zksync-era/blob/main/zk_toolbox/crates/zk_supervisor/src/commands/send_transactions/mod.rs

Probably we could unify it

Reply:
I dunno, what you introduced there is a serializable representation of transactions to send. Here I have a collection of hashes of sent transactions to await for.

Reply:
I think there should be a generic way to send and wait for l2 transactions (or even l1+l2).

@Deniallugo what's the vision there? Can @pompon0 use the logic in https://github.com/matter-labs/zksync-era/blame/main/zk_toolbox/crates/zk_supervisor/src/commands/send_transactions/mod.rs as is? In other words, what's the guideline to issue L2 transactions?

Reply:
I don't think that it's possible in a straight forward way, but i'll think how to improve it